Firestore/Example/Tests/Integration/FSTStreamTests.mm

```diff
@@ -327,4 +328,67 @@ - (void)testStreamCancelsIdleOnWrite {
   ]];
 }
 
+class MockCredentialsProvider : public firebase::firestore::auth::EmptyCredentialsProvider {
+ public:
+  MockCredentialsProvider() {
+    observed_refreshes_ = [NSMutableArray new];
+  }
+
+  void GetToken(bool force_refresh, firebase::firestore::auth::TokenListener completion) override {
+    [observed_refreshes_ addObject:[NSNumber numberWithBool:force_refresh]];
+    EmptyCredentialsProvider::GetToken(force_refresh, std::move(completion));
+  }
+
+  NSMutableArray<NSNumber *> *observed_refreshes() const {
+    return observed_refreshes_;
+  }
+
+ private:
+  NSMutableArray<NSNumber *> *observed_refreshes_;
```

Comment:
I made this an Objective-C class because I couldn't figure out if I could make a vector work with `XCTAssertEqualObjects`.

Reply:
I changed it to be track states, more in line with how the stream calls are tracked. Do you think it needs a comment, or is it similar enough to the rest of the file?